let DEBUG_MODE,ValidationError,CustomErrorHandler;_79a‍.x([["default",()=>_79a‍.o]]);_79a‍.w("../config",[["DEBUG_MODE",["DEBUG_MODE"],function(v){DEBUG_MODE=v}]]);_79a‍.w("joi",[["ValidationError",["ValidationError"],function(v){ValidationError=v}]]);_79a‍.w("../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);



const errorHandler = (err, req, res, next) => {
  let statusCode = 500;
  let data = {
    message: "Internal sever error",
    ...(DEBUG_MODE === "true" && {orignalError: err.message}),
  };

  if (err instanceof ValidationError) {
    statusCode = 422;
    data = {
      message: err.message,
    };
  }

  if (err instanceof CustomErrorHandler) {
    statusCode = err.status;
    data = {
      message: err.message,
    };
  }
  return res.status(statusCode).json(data);
};

_79a‍.d(errorHandler);
